app/views/lookup_keys/_fields.html.erb

```diff
@@ -1,59 +1,61 @@
-    %>
 
-    <div class="children_fields lookup_values">
-      <%= new_child_fields_template(f, :lookup_values, {:partial => "lookup_keys/value", :form_builder_attrs => {:is_param => is_param}}) %>
-      <%= f.fields_for :lookup_values do |lookup_values| %>
-        <%= render 'lookup_keys/value', :f => lookup_values, :is_param => is_param %>
-      <% end %>
-      <%= add_child_link '+ ' + _("Add Matcher-Value"), :lookup_values, { :title => _('add a new matcher-value pair')} %>
-      <%= documentation_button('4.2.6SmartMatchers') %>
-    </div>
+    <%= param_type_selector(f, :onchange => 'keyTypeChange(this)') %>
+    <%= textarea_f f, :default_value, :value => f.object.default_value_before_type_cast, :size => "col-md-8",
+                   :disabled                 => (f.object.is_param && (!f.object.override || f.object.use_puppet_default)),
+                   :rows                     => :auto, :help_inline => popover('', _("Value to use when there is no match")) %>
+    <%= checkbox_f(f, :use_puppet_default, :label => _('Use Puppet default'), :size => "col-md-8",
+                   :help_inline                   => use_puppet_default_help,
+                   :onchange                      => 'toggleUsePuppetDefaultValue(this, "default_value")',
+                   :disabled                      => (f.object.is_param && !f.object.override)) if is_param %>
+    </br>
+  </fieldset>
+  <div class="matcher-parent" <%= "id=#{(f.object.key || 'new_lookup_keys').to_s.gsub(' ', '_')}_lookup_key_override_value" %> style=<%= "display:none;" if (f.object.is_param && !f.object.override) %>>
+    <fieldset>
+      <legend><%= _("Prioritize Attribute Order") %></legend>
+      <h6><%= popover("", _("The order in which matchers keys are processed, first match wins.<br> You may use multiple attributes as a matcher key, for example, an order of <code>host group, environment</code> would expect a matcher such as <code>hostgroup = \"web servers\", environment = production</code>"),
+                      :title => _("The order in which values are resolved")).html_safe %> <%= _("Set the order in which values are resoved") %></h6>
+      <%= textarea_f f, :path, :rows => :auto, :label => _("Order"), :id => 'order', :size => "col-md-8", :onchange => 'fill_in_matchers()', :value => f.object.path %>
+    </fieldset>
+    <fieldset>
+      <legend><%= _("Specify Attribute Matcher") %>  <%= documentation_button('4.2.6SmartMatchers') %></legend>
+      <div class="children_fields lookup_values">
+        <%= render 'lookup_keys/values', :f => f, :is_param => is_param %>
+      </div>
+      <%= table_checkbox_f(f, :merge_overrides, :onchange => 'mergeOverridesChanged(this)',
```

Comment:
The position of these in the form is a bit lost, as they're stuck between a large "Add Matcher-Value" button and a heading, it took me a while to notice them.  It might be better at the top of this section above the table?